src/flows/pipes/MetricSelector/view.tsx

```diff
@@ -1,33 +1,48 @@
 // Libraries
-import React, {FC} from 'react'
+import React, {FC, useContext} from 'react'
 // Types
 import {PipeProp} from 'src/types/flows'
+import {Bucket} from 'src/types'
 
 // Contexts
 import BucketProvider from 'src/flows/context/buckets'
 import {SchemaProvider} from 'src/flows/pipes/MetricSelector/context'
+import {PipeContext} from 'src/flows/context/pipe'
 
 // Components
 import BucketSelector from 'src/flows/shared/BucketSelector'
 import FieldsList from 'src/flows/pipes/MetricSelector/FieldsList'
 import FilterTags from 'src/flows/pipes/MetricSelector/FilterTags'
 import SearchBar from 'src/flows/pipes/MetricSelector/SearchBar'
 
-const DataSource: FC<PipeProp> = ({Context}) => (
-  <BucketProvider>
-    <SchemaProvider>
-      <Context>
-        <div className="data-source--controls">
-          <div className="data-source--bucket">
-            <BucketSelector />
+const DataSource: FC<PipeProp> = ({Context}) => {
+  const {data, update} = useContext(PipeContext)
+  const updateBucket = (bucket: Bucket) => {
+    update({
```

Comment:
Does the `update` function send events or should we event bae this?

Reply:
didn't notice this got removed. thanks for flagging